licensing/src/main/java/io/crate/license/LicenseMetaData.java

```diff
@@ -0,0 +1,183 @@
+     *           "expirationDateInMs": 202020202,
+     *           "issuedTo": "organisation",
+     *           "signture": "XXX"
+     *       }
+     *     }
+     * </pre>
+     */
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(TYPE);
+        builder.field("expirationDateInMs", expirationDateInMs);
+        builder.field("issuedTo", issuedTo);
+        builder.field("signature", signature);
+        builder.endObject();
+        return builder;
+    }
+
+    public static LicenseMetaData fromXContent(XContentParser parser) throws IOException {
+        long expirationDateInMs = 0L;
+        String issuedTo = null;
+        String signature = null;
+        // advance from metadata START_OBJECT
+        XContentParser.Token token = parser.nextToken();
+        if (token != XContentParser.Token.FIELD_NAME || !Objects.equals(parser.currentName(), TYPE)) {
+            throw new LicenseMetadataParsingException("license FIELD_NAME expected but got " + parser.currentToken());
+        }
+        // advance to license START_OBJECT
+        if (parser.nextToken() != XContentParser.Token.START_OBJECT) {
+            throw new LicenseMetadataParsingException("license START_OBJECT expected but got " + parser.currentToken());
+        }
+
+        while ((token = parser.nextToken()) == XContentParser.Token.FIELD_NAME) {
+            if ("expirationDateInMs".equals(parser.currentName())) {
+                expirationDateInMs = parseLongField(parser);
+            } else if ("issuedTo".equals(parser.currentName())) {
+                issuedTo = parseStringField(parser);
+            } else if ("signature".equals(parser.currentName())) {
+                signature = parseStringField(parser);
+            } else {
+                throw new LicenseMetadataParsingException("unexpected FIELD_NAME " + parser.currentToken());
+            }
+        }
+        // license END_OBJECT is already consumed - check for correctness
+        if (parser.currentToken() != XContentParser.Token.END_OBJECT) {
+            throw new LicenseMetadataParsingException("expected the license object token at the end");
+
+        }
+        if (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            // each custom metadata is packed inside an object.
+            // each custom must move the parser to the end otherwise possible following customs won't be read
+            throw new LicenseMetadataParsingException("expected an object token at the end");
+        }
+        return new LicenseMetaData(expirationDateInMs, issuedTo, signature);
+    }
+
+    private static String parseStringField(XContentParser parser) throws IOException {
+        if (parser.nextToken() != XContentParser.Token.VALUE_STRING &&
+            parser.currentToken() != XContentParser.Token.VALUE_NULL) {
+            throw new LicenseMetadataParsingException("String value expected");
```

Comment:
The `text()`, `textOrNull` and `longValue()` methods already do some form of validation, not sure if we need these extra checks here?